--- a/meson.build
+++ b/meson.build
@@ -49,7 +49,7 @@
 include_dir = 'include/'
 include_dirs = include_directories(include_dir, '/usr/include/json-c/')
 
-scdoc = dependency('scdoc', version: '>=1.9.2', native: true)
+scdoc = dependency('scdoc', version: '>=1.9.2', native: true, required: false)
 if scdoc.found()
     scdoc_prog = find_program(scdoc.get_pkgconfig_variable('scdoc'), native: true)
     sh = find_program('sh', native: true)
