--- a/meson.build
+++ b/meson.build
@@ -156,7 +156,7 @@
 wayland_protos = dependency('wayland-protocols', version: '>=1.14')
 wayland_client = dependency('wayland-client')
 deps = [\
-    dependency('xcb'),
+    dependency('xcb', required: false),
     dependency('xkbcommon'),
     dependency('wayland-egl'),
     wayland_server,
@@ -164,7 +164,7 @@
     wayland_protos,
     dependency('pixman-1'),
     dependency('wlroots', version: ['>=0.15' ,'<0.17']),
-    dependency('x11'),
+    dependency('x11', required: false),
     dependency('json-c'),
     dependency('libnotify'),
     dependency('threads'),
